plugin_info/install.php

```diff
@@ -262,6 +262,33 @@ function convertBatteryStatus() {
 	jMQTT::logger('info', __("Commandes batterie définies directement sur les équipements jMQTT", __FILE__));
 }
 
+function moveCertsInDb() {
+	$oldPath = realpath(dirname(__FILE__) . '/../data/jmqtt/certs');
+	// for each brokers
+	foreach ((jMQTT::getBrokers()) as $broker) {
+		$broker->setConfiguration(jMQTT::CONF_KEY_MQTT_PROTO, boolval($broker->getConfiguration('mqttTls', 0)) ? 'mqtts' : 'mqtt');
+		$broker->setConfiguration('mqttTls', null);
+		$cert = $broker->getConfiguration('mqttTlsCaFile', '');
+		$broker->setConfiguration('mqttTlsCaFile', null);
+		if ($cert != '')
+			$broker->setConfiguration(jMQTT::CONF_KEY_MQTT_TLS_CA, file_get_contents($oldPath.'/'.$cert));
+		$cert = $broker->getConfiguration('mqttTlsClientCertFile', '');
+		$broker->setConfiguration('mqttTlsClientCertFile', null);
+		if ($cert != '')
+			$broker->setConfiguration(jMQTT::CONF_KEY_MQTT_TLS_CLI_CERT, file_get_contents($oldPath.'/'.$cert));
+		$cert = $broker->getConfiguration('mqttTlsClientKeyFile', '');
+		$broker->setConfiguration('mqttTlsClientKeyFile', null);
+		if ($cert != '')
+			$broker->setConfiguration(jMQTT::CONF_KEY_MQTT_TLS_CLI_KEY, file_get_contents($oldPath.'/'.$cert));
+		$broker->save();
```

Comment:
good idea for the migration, but there is a potential issue : certificate files can be binary encoded (DER) and so getting content and put it in configuration won't work.

Reply:
This should not be an issue as Paho MQTT only wants PEM files:
https://github.com/eclipse/paho.mqtt.python/blob/9782ab81fe7ee3a05e74c7f3e1d03d5611ea4be4/src/paho/mqtt/client.py#L750-L751
(That's why I took the liberty to directly transfert files content from config)

Reply:
Also I think that's a good way to ensure certificates are provided in PEM format by users.

Reply:
ok, perfect